lib/core-client/src/preview/parseArgsParam.ts

```diff
@@ -1,20 +1,34 @@
 import qs from 'qs';
 import { Args } from '@storybook/addons';
 import { once } from '@storybook/client-logger';
+import isPlainObject from 'lodash/isPlainObject';
 
 // Keep this in sync with validateArgs in @storybook/core
 const VALIDATION_REGEXP = /^[a-zA-Z0-9 _-]*$/;
-const validateArgs = (key = '', value: any = ''): boolean => {
-  if (key === null || value === null) return false;
+const validateArgs = (key = '', value: unknown): boolean => {
+  if (value === null || value === undefined) return true; // encoded as `!null` or `!undefined`
+  if (key === null) return false;
   if (key === '' || !VALIDATION_REGEXP.test(key)) return false;
+  if (typeof value === 'number') return true;
   if (typeof value === 'string') return VALIDATION_REGEXP.test(value);
-  if (Array.isArray(value)) return value.every((v) => validateArgs(key, v));
-  return Object.entries(value).every(([k, v]) => validateArgs(k, v));
+  if (Array.isArray(value)) return !value.some((v) => !validateArgs(key, v));
```

Comment:
I'm guessing this line was changed for performance reasons, so it short-circuits out on the first failure?

Reply:
I'm pretty sure the performance of `every(true)` is the same as `some(false)`, because both probably short circuit on the first failure

I honestly can't recall why I changed this. When I change it back the tests still pass, so I'll revert this.